cmd/podman/cp.go

```diff
@@ -223,8 +228,23 @@ func copy(src, destPath, dest string, idMappingOpts storage.IDMappingOptions, ch
 	copyWithTar := chrootarchive.CopyWithTarAndChown(chownOpts, digest.Canonical.Digester().Hash(), idMappingOpts.UIDMap, idMappingOpts.GIDMap)
 	untarPath := chrootarchive.UntarPathAndChown(chownOpts, digest.Canonical.Digester().Hash(), idMappingOpts.UIDMap, idMappingOpts.GIDMap)
 
-	if srcfi.IsDir() {
+	filename := filepath.Base(destPath)
+	if filename == "-" && !isFromHostToCtr {
+		file, err := os.Open(srcPath)
+		if err != nil {
+			return errors.Wrapf(err, "error opening file %s", srcPath)
+		}
+		defer file.Close()
+
+		data, err := ioutil.ReadAll(file)
```

Comment:
What if I'm copying a directory out of the container? I'd imagine we still need to tar it up.

Reply:
I'll fix this.